element_array_ephys/export/nwb.py

```diff
@@ -0,0 +1,384 @@
+from hdmf.backends.hdf5 import H5DataIO
+from nwb_conversion_tools.utils.genericdatachunkiterator import GenericDataChunkIterator
+from nwb_conversion_tools.utils.recordingextractordatachunkiterator import (
+    RecordingExtractorDataChunkIterator,
+)
+from pynwb import NWBHDF5IO
+from pynwb.ecephys import ElectricalSeries, LFP
+from pynwb.misc import Units
+from spikeextractors import (
+    SpikeGLXRecordingExtractor,
+    SubRecordingExtractor,
+    OpenEphysNPIXRecordingExtractor,
+)
+from tqdm import tqdm
+
+from element_session import session
+
+
+# from element_data_loader.utils import find_full_path
+
+
+class LFPDataChunkIterator(GenericDataChunkIterator):
+    """DataChunkIterator for LFP data that pulls data one channel at a time."""
+
+    def __init__(
+        self,
+        lfp_electrodes_query,
+        buffer_length: int = None,
+        chunk_length: int = 10000,
+    ):
+        self.lfp_electrodes_query = lfp_electrodes_query
+        self.electrodes = self.lfp_electrodes_query.fetch("electrode")
+
+        first_record = (
+            self.lfp_electrodes_query & dict(electrode=self.electrodes[0])
+        ).fetch(as_dict=True)[0]
+
+        self.n_channels = len(lfp_electrodes_query)
+        self.n_tt = len(first_record["lfp"])
+        self._dtype = first_record["lfp"].dtype
+
+        chunk_shape = (chunk_length, 1)
+        if buffer_length is not None:
+            buffer_shape = (buffer_length, 1)
+        else:
+            buffer_shape = (len(first_record["lfp"]), 1)
+        super().__init__(buffer_shape=buffer_shape, chunk_shape=chunk_shape)
+
+    def _get_data(self, selection):
+
+        electrode = self.electrodes[selection[1]][0]
+        return (self.lfp_electrodes_query & dict(electrode=electrode)).fetch1("lfp")[
+            selection[0], np.newaxis
+        ]
+
+    def _get_dtype(self):
+        return self._dtype
+
+    def _get_maxshape(self):
+        return self.n_tt, self.n_channels
+
+
+def check_module(nwbfile, name: str, description: str = None):
+    """
+    Check if processing module exists. If not, create it. Then return module.
+    Parameters
+    ----------
+    nwbfile: pynwb.NWBFile
+    name: str
+    description: str | None (optional)
+    Returns
+    -------
+    pynwb.module
+    """
+    assert isinstance(nwbfile, NWBFile), "'nwbfile' should be of type pynwb.NWBFile"
+    if name in nwbfile.processing:
+        return nwbfile.processing[name]
+    else:
+        if description is None:
+            description = name
+        return nwbfile.create_processing_module(name, description)
+
+
+def get_ephys_root_data_dir():
```

Comment:
We generally rely on a definition of the root directory in the workflow (aka `linking_module`). [The workflow definition](https://github.com/datajoint/workflow-array-ephys/blob/main/workflow_array_ephys/paths.py) gets pulled [here, when initializing the schema](https://github.com/datajoint/element-array-ephys/blob/main/element_array_ephys/ephys.py#L72). I wouldn't want to hardcode in the export process.

Is this right?
```suggestion
from ..ephys import get_ephys_root_data_dir
```